aten/src/ATen/native/vulkan/glsl/mean.glsl

```diff
@@ -2,36 +2,26 @@
 #define PRECISION $precision
 layout(std430) buffer;
 layout(std430) uniform;
-layout(set = 0, rgba16f, binding = 0) writeonly PRECISION uniform image3D uOutput;
-layout(set = 0, binding = 1) uniform PRECISION sampler3D uInput;
-layout(set = 0, binding = 2) uniform constBlock {
+layout(set = 0, binding = 0, rgba16f) uniform PRECISION restrict writeonly image3D   uOutput;
+layout(set = 0, binding = 1)          uniform PRECISION                    sampler3D uInput;
+layout(set = 0, binding = 2)          uniform                                        Block {
   int W;
   int H;
-  int OW;
-  int OH;
-}
-uConstBlock;
+} uBlock;
 
 layout(local_size_x_id = 1, local_size_y_id = 2, local_size_z_id = 3) in;
 
 void main() {
   ivec3 pos = ivec3(gl_GlobalInvocationID);
-  int W = uConstBlock.W;
-  int H = uConstBlock.H;
-  int OW = uConstBlock.OW;
-  int OH = uConstBlock.OH;
-  vec4 r = vec4(1.0) / float(W) / float(H);
+  vec4 r = vec4(1.0) / float(uBlock.W) / float(uBlock.H);
   vec4 acc = vec4(0);
   int xi, yi;
-  for (xi = 0; xi < W; ++xi) {
-    for (yi = 0; yi < H; ++yi) {
+  for (xi = 0; xi < uBlock.W; ++xi) {
+    for (yi = 0; yi < uBlock.H; ++yi) {
```

Comment:
Iterate over y in the outer loop, and x in the inner loop.  We are dealing with a texture that is packed in an opaque format, so this might not apply, but if and when the memory is laid our linearly that traversal has better locality of access.

Reply:
Check adaptive_avg_pool shader here: https://github.com/pytorch/pytorch/pull/47261/files